src/script/printer/python_doc_printer.cc

```diff
@@ -0,0 +1,105 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <tvm/runtime/registry.h>
+
+#include "./doc_printer.h"
+
+namespace tvm {
+namespace script {
+namespace printer {
+
+namespace {
+
+/*!
+ * \brief Print a Python literal string
+ *
+ * \param string the string to be printed
+ * \param out the output stream
+ */
+void PrintLiteralString(const String& string, std::ostringstream& out) {
+  // TODO(yelite): Escape and smart quote (choose ' or " automatically)
+  out << "\"" << string << "\"";
+}
+
+/*!
+ * \brief Print a tvm::ir::PrimExpr as Python literal
+ *
+ * This only supports IntImm and FloatImm with size of 64 bits
+ *
+ * \param expr the PrimExpr to be printed
+ * \param out the output stream
+ */
+void PrintLiteralPrimExpr(const PrimExpr& expr, std::ostringstream& out) {
+  const DataType& dtype = expr->dtype;
+
+  if (dtype == DataType::Int(64)) {
+    out << Downcast<IntImm>(expr)->value;
+  } else if (dtype == DataType::Float(64)) {
+    // TODO(yelite): make the float printing roundtrippable
+    std::ostringstream number_value;
+    number_value.precision(17);
+    number_value << Downcast<FloatImm>(expr)->value;
+    out << number_value.str();
+  } else if (dtype == DataType::Bool()) {
+    out << (Downcast<IntImm>(expr)->value ? "True" : "False");
+  } else {
+    LOG(FATAL) << "Cannot print value with dtype " << dtype << " as literal expression";
+  }
+}
+
+}  // namespace
+
+class PythonDocPrinter : public DocPrinter {
+ public:
+  PythonDocPrinter(const DocPrinterOptions& options) : DocPrinter(options) {}
+
+ protected:
+  using DocPrinter::PrintDoc;
+
+  void PrintTypedDoc(const LiteralDoc& doc) final;
+};
+
+void PythonDocPrinter::PrintTypedDoc(const LiteralDoc& doc) {
+  const ObjectRef& value = doc->value;
+  if (!value.defined()) {
+    output_ << "None";
+  } else if (const auto* expr_node = value.as<PrimExprNode>()) {
+    PrintLiteralPrimExpr(GetRef<PrimExpr>(expr_node), output_);
+  } else if (const auto* string_obj = value.as<StringObj>()) {
+    PrintLiteralString(GetRef<String>(string_obj), output_);
+  } else {
+    LOG(FATAL) << "Unsupported literal value type " << value->GetTypeKey();
+  }
+}
+
+std::unique_ptr<DocPrinter> GetPythonDocPrinter(const DocPrinterOptions& options) {
+  return std::make_unique<PythonDocPrinter>(options);
+}
+
+TVM_REGISTER_GLOBAL("script.printer.PrintDocAsPython")
```

Comment:
Let's think about API consistency between python and c++: if a public API exists in c++, then it should exist also in python, and vice versa. This is a strict standard in non-legacy TVM codebase (e.g. TIR scheduling, MetaSchedule)

In our particular case, I would propose:
- renaming from the `PrintDocAsPython`, use `DocToScriptPython` as the public C++ / python API
- `GetPythonDocPrinter` is Relatively not super useful (only used by the entry function of TVMScript printer) and there is no actual need to expose it in any header file (correct me if I was wrong)